astropy/cosmology/units.py

```diff
@@ -356,7 +379,7 @@ def with_redshift(
 # ===================================================================
 
 
-def with_H0(H0=None):
+def with_H0(H0: Quantity | None = None) -> Equivalency:
```

Comment:
Why don't you add the physical type of `H0` to its annotation?

Reply:
Stripped by ruff.

Reply:
It would be very helpful to have the physical type of the `Quantity` in the annotation, but if it's not simple to include then it shouldn't hold back this pull request.